cassandra/cqltypes.py

```diff
@@ -660,6 +660,24 @@ def serialize(val, protocol_version):
         return int64_pack(nano)
 
 
+class DurationType(_CassandraType):
+    typename = 'duration'
+
+    @staticmethod
+    def deserialize(byts, protocol_version):
+        months, days, nanoseconds = vints_unpack(byts)
+        return util.Duration(months, days, nanoseconds)
+
+    @staticmethod
+    def serialize(duration, protocol_version):
+        try:
+            duration.validate()
```

Comment:
I think it is better to keep it since a user could modify the duration instance manually by accessing its attribute.